src/wasm/wasm.cpp

```diff
@@ -1541,10 +1548,16 @@ void ResumeThrow::finalize() {
   if (handleUnreachableOperands(this)) {
     return;
   }
+  if (cont->type.isNull()) {
+    // This will never be executed and the instruction will not be emitted.
+    // Model this with an uninhabitable cast type.
+    type = cont->type.with(NonNullable);
```

Comment:
`ResumeThrow` should be typed exactly like `Resume`, so this is not correct, either.